packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -447,26 +539,39 @@ export class JobDefinition extends Resource implements IJobDefinition {
           ? this.buildLogConfigurationSecretOptions(container.logConfiguration.secretOptions)
           : undefined,
       } : undefined,
-      memory: container.memoryLimitMiB || 4,
       mountPoints: container.mountPoints,
       privileged: container.privileged || false,
-      resourceRequirements: container.gpuCount
-        ? [{ type: 'GPU', value: String(container.gpuCount) }]
-        : undefined,
+      networkConfiguration: container.networkConfiguration ? {
+        assignPublicIp: container.networkConfiguration.assignPublicIp ? 'ENABLED' : 'DISABLED',
+      } : undefined,
       readonlyRootFilesystem: container.readOnly || false,
       ulimits: container.ulimits,
       user: container.user,
-      vcpus: container.vcpus || 1,
       volumes: container.volumes,
+      fargatePlatformConfiguration: container.fargatePlatformConfiguration ? {
+        platformVersion: container.fargatePlatformConfiguration.platformVersion,
+      } : undefined,
+      ...(isFargate ? {
+        resourceRequirements: [
+          { type: 'VCPU', value: String(container.vcpus || 0.25) },
+          { type: 'MEMORY', value: String(container.memoryLimitMiB || 512) },
+        ],
+      } : {
+        memory: container.memoryLimitMiB || 4,
+        vcpus: container.vcpus || 1,
```

Comment:
From CloudFormation docs: 

> For Fargate resources, you should specify the memory requirement using resourceRequirement. You can also do this for EC2 resources.

If we can use `resourceRequirements` for both - lets just do that and avoid this code branch.

Reply:
Done